paywall/src/utils/setUserMetadata.ts

```diff
@@ -0,0 +1,27 @@
+import { WalletService } from '@unlock-protocol/unlock-js'
+import { UserMetadata } from '../unlockTypes'
+import configure from '../config'
+
+export function setUserMetadata(
+  lockAddress: string,
+  userAddress: string,
+  metadata: UserMetadata,
+  callback: (error: any, value: any) => void
+) {
+  const { unlockAddress, locksmithUri } = configure()
+
+  // TODO: investigate having a singleton WalletService to avoid any
+  // instantiation overhead if we start to use it more frequently
+  const walletService = new WalletService({ unlockAddress })
```

Comment:
I think this migt require to call `.connect` because that's how the provider object is set and the provider is then used by `walletService.setUserMetadata` to sign?